enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/pipes/ProjectionSlottedPipe.scala

```diff
@@ -39,11 +39,16 @@ case class ProjectionSlottedPipe(source: Pipe, introducedExpressions: Map[Int, E
         val result = expression(ctx, state)
         ctx.setRefAt(offset, result)
   }
+
   protected def internalCreateResults(input: Iterator[ExecutionContext], state: QueryState): Iterator[ExecutionContext] = {
-    input.map {
-      ctx =>
-        projectionFunctions.foreach(_(ctx, state))
-        ctx
+    if (projectionFunctions.isEmpty)
```

Comment:
This is weird. Why would we end up with a projection operator that does not project? I much prefer if we simply got rid of the plan instead.

Reply:
Or not build the pipe. This is a quick fix for the observation that we get a lot of empty projections from the rewriters around ORDER BY and RETURN. We should solve that, but do you prefer to have this problem until then?